api/channel.go

```diff
@@ -397,10 +397,15 @@ func updateChannelPurpose(c *Context, w http.ResponseWriter, r *http.Request) {
 }
 
 func getChannels(c *Context, w http.ResponseWriter, r *http.Request) {
+	// check if the team exist
+	if result := <-Srv.Store.Team().Get(c.TeamId); result.Err != nil {
```

Comment:
the issue is not only if it is nil but also a wrong teamId. 
for example if you send a wrong team id and you have DM channels it will return the DM channel and not say the team id is invalid.